### PR TITLE
Efss backend fixes

### PR DESCRIPTION
**Pull request by michielbdejong**
Continuation of https://github.com/cs3org/reva/pull/3524

### CODE REVIEW

**Comment by vascoguita** [comment]
Thanks for creating a PR from your fork. This way the CI will run even faster since more runners are available to work in parallel (we just have 8 self-hosted runners so far).
@michielbdejong can I delete the cs3org/reva:efss-backend-fixes branch?

File: changelog/unreleased/improve-ocmd-error-logs.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Enhancement: Improve error logging in ocmd flow
+
+https://github.com/cs3org/reva/pull/3524
```
Comment (vascoguita) [inline changelog/unreleased/improve-ocmd-error-logs.md:3]: To pass the changeling CI, just update this line to https://github.com/cs3org/reva/pull/3524

**Comment by michielbdejong** [comment]
@gmgigi96 it's green now :)

File: pkg/ocm/share/manager/json/json.go [hunk 2]
```diff
@@ -264,6 +265,15 @@ func (m *mgr) Share(ctx context.Context, md *provider.ResourceId, g *ocm.ShareGr
...
 		}
 
+		log := appctx.GetLogger(ctx)
+		log.Info().Msg("pkg/ocm/share/manager/json calls sender.Send")
+		log.Info().Msgf("pkg/ocm/share/manager/json shareWith: %s", g.Grantee.GetUserId().OpaqueId)
+		log.Info().Msgf("pkg/ocm/share/manager/json name: %s", name)
+		log.Info().Msgf("pkg/ocm/share/manager/json providerId: %s", fmt.Sprintf("%s:%s", md.StorageId, md.OpaqueId))
+		log.Info().Msgf("pkg/ocm/share/manager/json owner: %s", userID.OpaqueId)
+		log.Info().Msgf("pkg/ocm/share/manager/json protocol: %s", protocol)
+		log.Info().Msgf("pkg/ocm/share/manager/json meshProvider: %s", userID.Idp)
```
Comment (gmgigi96) [inline pkg/ocm/share/manager/json/json.go:275]: These seems to be debug logs, can you remove them? Or log them better?

File: pkg/ocm/share/manager/nextcloud/nextcloud.go [hunk 3]
```diff
@@ -342,18 +357,45 @@ func (sm *Manager) Share(ctx context.Context, md *provider.ResourceId, g *ocm.Sh
 				"options": map[string]string{
 					"permissions":  pm,
 					"sharedSecret": token,
+					"remote":       sm.webDAVHost,
 				},
 			}
 		}
+
+		log := appctx.GetLogger(ctx)
+		log.Info().Msg("pkg/ocm/share/manager/nextcloud calls sender.Send")
```
Comment (gmgigi96) [inline pkg/ocm/share/manager/nextcloud/nextcloud.go:366]: Same as above

File: pkg/ocm/share/manager/nextcloud/nextcloud.go [hunk 4]
```diff
@@ -498,33 +543,32 @@ func (sm *Manager) ListReceivedShares(ctx context.Context) ([]*ocm.ReceivedShare
...
-					Owner:   altResultShare.Owner,
-					Creator: altResultShare.Creator,
-					Ctime:   altResultShare.Ctime,
-					Mtime:   altResultShare.Mtime,
-				},
-				State: respArr[i].State,
-			}
+			return nil, errors.New("received an unreadable share object from the EFSS backend")
+		}
+		pointersBaseShare[i] = &ocm.Share{
```
Comment (gmgigi96) [inline pkg/ocm/share/manager/nextcloud/nextcloud.go:554]: use the append function to add the struct to the list

**Review by gmgigi96 — CHANGES_REQUESTED** [review]

**Pull request merged**